### PR TITLE
[ADVAPP-1]: Enforce License Rules for Emails

### PR DESCRIPTION
**Pull request by Orrison**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-1

### Technical Description

Enforces License Rules for emails, we now track usage quota per Outbound Deliverable that gets sent to SES. Before sending we calculate how many "recipients" there will be for a single Mail item and count that towards an `OutboundDeliverables` quota usage.

### Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Content or styling update (Changes which don't affect functionality)
- [ ] DevOps
- [ ] Documentation

### Any deployment steps required?

A deployment step could be a command that needs to be executed or an ENV key that needs to be added, for example.

- [ ] Yes, please specify
- [x] No

_______________________________________________

## Before contributing and submitting this PR, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/canyongbs/advisingapp/blob/main/README.md#contributing).
* [x] Title the PR with the ticket/issue number and a short description of the changes made. Or if no ticket/issue exists, title the PR with a short description of the changes made
* [x] Linked a relevant ticket or issue or describe the issue/feature which this PR resolves/implements.
* [x] Resolved all conflicts, if any.
* [x] Rebased your branch PR on top of the latest upstream `develop` branch.

### CODE REVIEW

File: app-modules/notification/src/DataTransferObjects/EmailChannelResultData.php [hunk 1]
```diff
@@ -42,7 +42,10 @@ class EmailChannelResultData extends NotificationResultData
 {
     public function __construct(
         public bool $success,
+        public array $recipients = [],
         public ?SentMessage $message = null,
         public ?string $error = null,
-    ) {}
+    ) {
+        parent::__construct();
```
Comment (dgoetzit) [inline app-modules/notification/src/DataTransferObjects/EmailChannelResultData.php:49]: Just for my edification, why is this necessary?

File: app-modules/notification/src/Notifications/Channels/EmailChannel.php [hunk 2]
```diff
@@ -81,11 +108,32 @@ public static function afterSending(object $notifiable, OutboundDeliverable $del
...
+        }
+
+        // 1 for the primary recipient, plus the number of cc and bcc recipients
+        $estimatedQuotaUsage = 1 + count($notification->toMail($notifiable)->cc) + count($notification->toMail($notifiable)->bcc);
+
+        $licenseSettings = app(LicenseSettings::class);
+
+        $resetWindow = $licenseSettings->data->limits->getResetWindow();
+
+        $currentQuoteUsage = OutboundDeliverable::where('channel', 'email')
```
Comment (dgoetzit) [inline app-modules/notification/src/Notifications/Channels/EmailChannel.php:133]:
```suggestion
        $currentQuotaUsage = OutboundDeliverable::where('channel', 'email')
```

**Review by dgoetzit — APPROVED** [review]
Looks good - just had a question and variable change.

File: app-modules/notification/src/DataTransferObjects/EmailChannelResultData.php (hunk 1 above)
Comment (Orrison) [reply app-modules/notification/src/DataTransferObjects/EmailChannelResultData.php:49]: It's just best practice for an extension to call `parent::__construct();` to make sure the object is being instantiated correctly. It's not always needed and there are times when maybe you wouldn't want it. But PHPStorm and some static analysis will yell at you without.

**Review by dgoetzit — APPROVED** [review]
